### DISCUSSION TITLE
zsh: killed ./app

### DISCUSSION THREAD

**Discussion by d0x7** [pocketbase/pocketbase#5168] [Q&A] [answered]
Hey, I've got this really weird issue, but I'm not sure if it's enough to actually open an issue, so I wanted to ask here first.

Basically, when building my app (thats using pocketbase) and then running said executable, I get "zsh: killed ./app". I have zero clue what's the issue here. It does not occur when using `go run`, only when executing a binary built using `go build`. I tried narrowing the issue down and it also happens with the most minimal setup I can imagine:

```go
package main

import (
	"fmt"
	"github.com/pocketbase/pocketbase"
)

func main() {
	fmt.Println("Setting up Pocketbase..")
	_ = pocketbase.New()
	fmt.Println("Done setting up pocketbase!")
}
```

If I remove the `pocketbase.New` call, the app executes just fine.
I'm on macOS 15, using Go 1.22.5 and, well, zsh as a shell.

I thought it's maybe related to go and the new macOS version, but I don't think I came across a different app/library that has the same issue (yet), so that's why I'm asking here if anyone has got an idea maybe.

**Comment by ganigeorgiev** [comment]
> If I remove the pocketbase.New call, the app executes just fine.

I don't understand what do you mean by this but the above code sample is incomplete.

`pocketbase.New()` just creates an app instance, it doesn't start a web server. You need to call `Start()` on the returned app instance to start the web server (aka. the task that will block your process from exiting).

See the minimal Go example in https://pocketbase.io/docs/go-overview/.

If you need further help, please elaborate a little more on your setup and use case to understand the bigger picture and the actual problem.

**Reply by ganigeorgiev** [reply, marked as answer]
By googling `zsh: killed` this seems to be a common issue and could be returned for a lot of reasons, including permission issues or missing C tooling dependencies if you have `CGO_ENABLED=1` by default.

This wouldn't explain why it works in your case with `go run` and not `go build`, but just in case to rule out CGO related issue/permission, try to build with `CGO_ENABLED=0` so that it generates a statically linked executable:

```sh
CGO_ENABLED=0 go build
```

If that still doesn't work, then you could try with a different shell to confirm that it is something with zsh only.

**Reply by d0x7** [reply]
I know that I need to call `Start()` to actually start Pocketbase, it's just that the most minimal version to produce this error is even just creating an Pocketbase instance by `Pocketbase.New`, even if I don't do anything with it. If I leave this line out and have just the `fmt.Println` calls, I can run the executable without issues that's what I meant by removing that line.

I also googled a bit and it seems you're on the right path. When disabling CGO, it works and I can execute it - thanks for that already!

But it seems to be a deeper issue related somewhere within gopls and code signing. Only thing I don't understand is why I can execute the binary when the code is only some print statements, but the issue occurs when I put `pocketbase.New` in there. I'm going to sleep now, but I'll be checking in later what's up with this issue. But I'm very sure this issue has nothing to do with Pocketbase itself - so thanks anyways!

Here's some related issues if anyone's interested: 
https://github.com/golang/go/issues/65810#issuecomment-1954275502
https://github.com/golang/go/issues/63997#issuecomment-1847840272
https://github.com/golang/go/issues/64351
https://github.com/golang/go/issues/63997
